odoo/tools/mail.py

```diff
@@ -493,6 +493,12 @@ def generate_tracking_message_id(res_id):
     rndstr = ("%.15f" % rnd)[2:]
     return "<%s.%.15f-openerp-%s@%s>" % (rndstr, time.time(), res_id, socket.gethostname())
 
+def email_split_keep_invalid(text):
+    """Split email addresses like email_split but do not remove invalid/falsy values."""
+    if not text:
+        return []
+    return [addr for name, addr in getaddresses([text])]
+
 def email_split_tuples(text):
```

Comment:
I'll can add ``email_split_tuples_keep_invalid`` as a wrapper around ``getaddresses`` maybe in case we use something else in the future? But ``split_tuples`` and ``split_keep_invalid`` are not compatible (one filters the result, the other splits the result)

Reply:
can't you do something like this ?
```python
def email_split_keep_invalid(text):
    if not text:
        return []
    return [addr for name, addr in getaddresses([text])]


def email_split_tuples(text):
    address = email_split_keep_invalid(text)
    return [
        addr for addr in address
        if addr[1] and '@' in addr[1]
    ]
```

they look very similar

Reply:
The issue is ``split_tuples`` actually returns both ``name`` and ``addr`` (e.g. ['john', 'john@test.lan']). Since ``split`` and ``split_keep_invalid`` remove that part, I can't go from one to the other.